### PR TITLE
UHM-3485- Modify Obs Component to handle date times

### PR DESCRIPTION
**Pull request by rhenshaw56**

### CODE REVIEW

File: src/components/widgets/FieldInput.js [hunk 1]
```diff
@@ -1,43 +1,31 @@
 import React from 'react';
-import PropTypes from 'prop-types';
```
Comment (mogoodrich) [inline src/components/widgets/FieldInput.js]: Same here, I think this is inadvertently being overwritten?

File: src/components/widgets/Loader.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import '../../../assets/css/loader.css';
+
+const Loader = () => (
+  <div className="custom-loader">
```
Comment (mogoodrich) [inline src/components/widgets/Loader.js]: What does the Loader component do?

File: src/features/form/validations.js [hunk 3]
```diff
@@ -9,9 +11,16 @@ const abnormalMaxValue = max => value =>
 
 const abnormalMinValue = min => value => (value && value < min ? `Abnormal value` : undefined);
 
+const dateToInt = dateStr => new Date(dateStr).getTime();
+
+const maxDateValue = maxDate => value =>
+  value && (dateToInt(value)) > dateToInt(maxDate) ? `Date cannot be in the future` : undefined;
```
Comment (mogoodrich) [inline src/features/form/validations.js]: It looks like this is a generic validator that supports testing against any specific maxDate?  This is good, but the validation message is inaccurate in this case (always says "cannot be in the future")

File: src/components/form/Obs.jsx [hunk 4]
```diff
@@ -57,27 +56,7 @@ const Obs = (props) => {
       />
     );
   }
-};
-
-Obs.propTypes = {
```
[the remark this answers is not in the document]
Comment (rhenshaw56) [reply src/components/form/Obs.jsx]: @mogoodrich you're right, must have been when I rebased! Will update it back. Thanks

File: src/features/form/validations.js [hunk 5]
```diff
@@ -9,9 +11,16 @@ const abnormalMaxValue = max => value =>
 
 const abnormalMinValue = min => value => (value && value < min ? `Abnormal value` : undefined);
 
+const dateToInt = dateStr => new Date(dateStr).getTime();
+
+const maxDateValue = maxDate => value =>
+  value && (dateToInt(value)) > dateToInt(maxDate) ? `Date should be earlier or equal To Today Date` : undefined;
```
Comment (mogoodrich) [inline src/features/form/validations.js]: Shouldn't this really be something like '{maxDate}'?

**Pull request merged**